bftengine/src/bftengine/messages/MessageBase.hpp

```diff
@@ -111,6 +129,21 @@ class MessageBase {
 
   static constexpr uint64_t SPAN_CONTEXT_MAX_SIZE{1024};
   static constexpr uint32_t MAX_BATCH_SIZE{1024};
+
+  bool isIncomingMsg_ = false;
+
+  // Static data structures gathering statistics for diagnostics server:
+
+  // although MsgCode::MaxMsgCodeVal is much higher than the actual number of real message types,
+  // we allocate an array of size MsgCode::MaxMsgCodeVal for simplicity and avoiding using locks,
+  // as this structures are a shared resource between threads.
+  static std::array<std::atomic<size_t>, MsgCode::MaxMsgCodeVal> AliveIncomingExtrnMsgsBufs;
```

Comment:
you have

-  5 static member variables (protected)
- 5 static public member functions (public)
- all are dedicated for a single purpose - statistics
- they are scattered around a parent class.

I think you should wrap them in a static sub class Stats  (etter not remind the diag server, maybe others want the info in the future? maube metrics? who knows.) for better encupsulation and code organization.

Reply:
will encapsulated in a struct.